UCLALibrary/LBS
# SYS-1420: Merge Excel data into reporting table

Implements [SYS-1420](https://uclalibrary.atlassian.net/browse/SYS-1420). Internal only, not tagged for deployment.

This PR builds / updates the data used by the reports:
* New `LibraryData` model corresponding to the original `Master G&E Report` Access table
* New mapping table to allow initial import of `LibraryData` via Excel file during development
* New fixture `library_data.json` to automatically load `LibraryData` for development, via `docker_scripts/entrypoint.sh`
* New methods in `views_utils.py`
  * `add_funds()`: Adds new `LibraryData` instances (if any) from imported campus data
  * `update_data()`: Merges campus data and updates `LibraryData` instances to final state for reporting
  * For now, these new methods are called automatically after campus Excel files are loaded.
  * Please ignore the `print` statements in the code; they'll be replaced with proper logging later.

Other changes:
* Added `help_text` to the upload form, with basic info for campus Excel files
* Added `docker-compose_PGADMIN.yml` to allow use of `pgAdmin` (via container) during development
* Added basic documentation about using `pgAdmin` to `README.md`

### Testing
This PR does **not** add any tests for new code, because (a) that would delay this PR even more, and (b) I'm only moderately confident the code is correct.  

For now, test manually:
1. Restart application to add new model and load data
2. Go to http://127.0.0.1:8000/ge/report/ and sign in as `admin/admin` if needed
3. Upload the 3 full campus files emailed by LBS on Oct 12 2023
4. If all goes well, after about 10 seconds these messages should appear:
```
All data was imported successfully.
Reports are being generated.
```

Django logs (`docker-compose logs`) should include messages like these:
```
django_1  | Warning: No matching BFS (consolidated) data found for new_fund.fau_fund='36688'
django_1  | qryAAA_0Clear: 607 updated
django_1  | qryAAA_1UpdateMTF: 296 updated
django_1  | qryAAA_2ProjIncomFound: 297 updated
django_1  | qryAAA_2ProjIncomReg: 62 updated
django_1  | qryAAA_3FoundTotVal: 176 updated
django_1  | qryAAA_3FoundTotVal_2: 210 updated
django_1  | qryAAA_3FoundTotVal_3: 0 updated
django_1  | qryAAA_3RegTotVal: 111 updated
django_1  | qryAAA_3RegTotVal_2: 62 updated
django_1  | qryAAA_3RegTotVal_3: 41 updated
django_1  | qryAAA_3RegTotVal_4: 22 updated
django_1  | qryAAA_5_5400: 449 updated
django_1  | qryAAA_6TotalBalance: 607 updated
```
